### DISCUSSION TITLE
How to create a project which outputs, WASM, IOS and Android without manual reconfiguring.

### DISCUSSION THREAD

**Discussion by CJBuchel** [fzyzcjy/flutter_rust_bridge#2268] [Q&A] [answered]
I have a project that will be compiled for WASM, IOS and Android.
WASM and Android was fairly simple because I just needed to use rlib/cdylib in my cargo.toml and compile normally when it's used as a dependency in my rust project.

Or the regular code generator using `flutter_rust_bridge_codegen generate`, `flutter_rust_bridge build-web` and `cargo ndk -o ../android/app/src/main/jniLibs build`

But it breaks down pretty quickly when i need to do IOS as well as it needs `staticlib` instead of `cdylib`. Personally I'm trying to avoid manually configuring it, instead getting my ci-cd pipelines to manage generating and outputting the final generated code and binaries from my crate.

To get around this I tried the messy style of creating 3 libraries, my original `tms-infra` which compiles just to rlib that has my actual code, along with the FRB functions and structures. Then two more `tms-infra-dynamic` and `tms-infra-static` which import the `tms-infra` library and compile to their cdylib and static counter parts. And they're effectively empty other than just publically exposing the originally library in their main `lib.rs` file via `pub use tms_infra::*;`

Which *should* expose the underlying crate `crate::infra` which I'm using in my projects.
Sadly this weird method doesn't work when I try to generate any code.
`flutter_rust_bridge_codegen generate --dart-entrypoint-class-name $TMS_FLUTTER_ENTRY_NAME --rust-input crate::infra --rust-root tms-infra-dynamic --dart-output $TMS_FLUTTER/lib/generated`

the output generated folder is pretty much empty other than the 3 frb_generated dart files. Is there a way to expose the FRB segments from the underlying `tms-infra` library so that the code generator will pick it up and output correctly. Or better, a nicer method of managing needing to compile to both `cdylib` and `staticlib` without needing manual intervention

**Comment by fzyzcjy** [comment]
> But it breaks down pretty quickly when i need to do IOS as well as it needs staticlib instead of cdylib

I remembered my setup was like `change_code_to_cdylib && frb generate && change_code_back` script or something like that. It is ugly, but until Rust provides a good way to configure staticlib/cdylib/etc I guess this may be a workaround...

**Reply by CJBuchel** [reply, marked as answer]
Yea, I went through a bunch of existing issues on github for rust hoping to find a nice solution.
There's been a lot of talk for target based crate types. Modifying the conditions and crate-type based on the target, but... it's been up in the air for years. Doubt they'll ever settle on doing that now.

Currently my workaround in the pipelines is yea, to just replace the "cdylib" with "staticlib" using a script before compiling to ios.
```yaml
# run ios code gen
  - ${{ if eq(parameters.target, 'ios') }}:
    - script: |
        cd tms-infra
        perl -pi -e 's/"cdylib"/"staticlib"/g' Cargo.toml
        cat Cargo.toml
        cargo build --release --target aarch64-apple-ios
        cp ../target/aarch64-apple-ios/release/libtms_infra.a $(Build.SourcesDirectory)/tms-client/ios/Runner
      displayName: 'Run iOS Codegen'
```

It's not great, but it works.

**Reply by fzyzcjy** [reply]
I see... Looks like this may be the current way
